Reqnroll.Tools.MsBuild.Generation/build/Reqnroll.Tools.MsBuild.Generation.targets

```diff
@@ -92,6 +92,7 @@
 
     <GenerateFeatureFileCodeBehindTask
       ProjectPath="$(MSBuildProjectFullPath)"
+      IntermediateOutputPath="$(IntermediateOutputPath)"
```

Comment:
The MSBuild way of doing this would have been to compose the metadata _before_ passing the items to the task, doing all the path manipulation by setting a property in the item group like:

```xml
<ItemGroup>
    <ReqnrollFeatureFiles Include="**/*.feature">
        <GherkinMessagesResource>$(IntermediateOutputPath)%(SourceFiles.Identity)</GherkinMessagesResource>
    </ReqnrollFeatureFile>
</ItemGroup>
```

This makes it a lot easier to understand how these items are mapped 1:1 and will enable us to use these properties in the target inputs.

**Don't change this implementation now, it's good enough to do what we need today.** I'd like to revisit the MSBuild integration as a follow-up, keeping this in mind.

Reply:
OK. Let me know when you work on the overall MSBuild structure. I can assist with replumbing the metadata to messages generator.